In [ ]:
import numpy as np
from utilities import *


# Use os.path.join() to create the correct file path
filepath = os.path.join(get_filepath(), "Intra", "train", "rest_105923_1.h5")

# Load the data
data = read_h5py_file(filepath)



<class 'numpy.ndarray'>
(248, 35624)


The files have the following format: “taskType subjectIdentifier number.h5”
where taskType can be rest, task motor, task story math, and task working memory.
In practice, these tasks correspond to the activities performed by the subjects:

• Resting Task: Recording the subjects’ brain while in a relaxed resting
state.

• Math & Story Task : Subject performs mental calculation and language
processing task.

• Working Memory task: Subject performs a memorization task.

• Motor Task : Subject performs a motor task, typically moving fingers or feets.

The subject identifier is made of 6 numbers, and the number at the end cor-
responds to a chunk part. This number has no particular meaning (splitted
files are easier to handle in terms of memory management). The folder “In-
tra” contains the files of 1 subject only. In the folder “Cross”, 2 subjects are
contained in the train folder while the 3 test folders contain different subjects
from the ones contained in the train folder. As seen in the section above, each
2 file is represented by a matrix of shape 248 x 35624. The number of rows, 248,
corresponds to the number of magnetometer sensors placed on the human scalp.
The number of columns, 35624, corresponds to the time steps of a recording.